core/src/main/resources/org/apache/spark/ui/static/stagepage.js

```diff
@@ -0,0 +1,468 @@
+                            }
+                        },
+                        {
+                            data : function (row, type) {
+                                return row.shuffleReadRecords != 0 ? formatBytes(row.shuffleRead/row.shuffleReadRecords) : 0;
+                            }
+                        },
+                        {
+                            data : function (row, type) {
+                                return row.shuffleWriteRecords != 0 ? formatBytes(row.shuffleWrite/row.shuffleWriteRecords) : 0;
+                            }
+                        },
+                        {
+                            data : function (row, type) {
+                                return typeof row.memoryBytesSpilled != 'undefined' ? formatBytes(row.memoryBytesSpilled) : "";
+                            }
+                        },
+                        {
+                            data : function (row, type) {
+                                return typeof row.diskBytesSpilled != 'undefined' ? formatBytes(row.diskBytesSpilled) : "";
+                            }
+                        }
+                    ],
+                    "order": [[0, "asc"]]
+                }
+                $(tasksSummarytable).DataTable(task_summary_conf);
+                $('#active-tasks [data-toggle="tooltip"]').tooltip();
+
+                // building accumulator update table
+                var accumulatorTable = "#accumulator-table";
+                var accumulator_conf = {
+                    "data": accumulator_table,
+                    "columns": [
+                        {data : "id"},
+                        {data : "name"},
+                        {data : "value"}
+                    ],
+                    "paging": false,
+                    "searching": false,
+                    "order": [[0, "asc"]]
+                }
+                $(accumulatorTable).DataTable(accumulator_conf);
+
+                // building tasks table
+                var taskTable = "#active-tasks-table";
+                var task_conf = {
+                    "data": task_table,
+                    "columns": [
+                        {data: function (row, type) {
+                            return type !== 'display' ? (isNaN(row.index) ? 0 : row.index ) : row.index;
+                            }
+                        },
+                        {data : "taskId"},
+                        {data : "attempt"},
+                        {data : "taskState"},
+                        {data : "taskLocality"},
+                        {
+                            data : function (row, type) {
+                                return row.executorId + ' / ' + row.host;
+                            }
+                        },
+                        {data : "executorLogs", render: formatLogsCells},
+                        {data : "launchTime"},
```

Comment:
The `formatDate` function in `historypage.js` should be moved to `utils.js` and used here

Reply:
Fixed it